### PR TITLE
Feature/fix sharp edges

### PR DESCRIPTION
**Pull request by marcrecurve**
Integrating some of the bug fixes we found yesterday, @philngo 

Do you have some other way of testing this stuff? It runs cleanly with my small local data set. Haven't tested on GCS.

### CODE REVIEW

**Comment by philngo** [comment]
We'll have to set up some testing buckets or something to test that storage client.

File: oeem_etl/storage.py [hunk 1]
```diff
@@ -49,11 +49,11 @@ def _make_target_classes(self):
 
         class TargetWithClient(target):
             def __init__(self, path):
-                super().__init__(path, client=client, format=MixedUnicodeBytesFormat())
+                super(TargetWithClient, self).__init__(path, format=MixedUnicodeBytesFormat())
```
Comment (philngo) [inline oeem_etl/storage.py]: GCS and S3 will both need the client keyword arg, even though the LocalTarget doesn't need it, so we need to manage those cases as well here and when making the flag target below.

Comment (marcrecurve) [inline oeem_etl/storage.py]: Ah, yes. Forgot about that detail.

**Comment by philngo** [comment]
For the short term, since we don't have CI set up on this repo and some parts are fairly tricky to test, I can also pull in your changes and try things out locally.

**Comment by philngo** [comment]
This looks good to me

**Pull request merged**